derp/derp_client.go

```diff
@@ -220,15 +210,16 @@ func (c *Client) send(dstKey key.Public, pkt []byte) (ret error) {
 	c.wmu.Lock()
 	defer c.wmu.Unlock()
 	if c.rate != nil {
-		pktLen := frameHeaderLen + len(dstKey) + len(pkt)
+		pktLen := frameHeaderLen + 32 + len(pkt)
 		if !c.rate.AllowN(time.Now(), pktLen) {
 			return nil // drop
 		}
 	}
-	if err := writeFrameHeader(c.bw, frameSendPacket, uint32(len(dstKey)+len(pkt))); err != nil {
+
+	if err := writeFrameHeader(c.bw, frameSendPacket, uint32(32+len(pkt))); err != nil {
```

Comment:
Ditto

Reply:
Resolving to keep conversation about API in prior comment. Resolution here is same as abover.